OpenRA.Mods.Common/AI/States/GroundStates.cs

```diff
@@ -71,7 +78,7 @@ public void Tick(Squad owner)
 
 			if (!owner.IsTargetValid)
 			{
-				var closestEnemy = owner.Bot.FindClosestEnemy(owner.Units.Random(owner.Random).CenterPosition);
+				var closestEnemy = FindClosestEnemy(owner);
```

Comment:
Note that this changes `.Random` to `.First` which is a behaviour change, not just a refactor/perf fix.

I'm hopeful that this is a *good thing* in that it might fix the bug where an AI squad will oscillate between two targets without ever moving to attack either.  This will require some careful testing to qualify.